packages/dara-core/dara/core/internal/encoder_registry.py

```diff
@@ -101,6 +105,27 @@ def _get_pandas_array_encoder(array_type: Type[Any], dtype: Any, raise_: bool =
     pandas.Timestamp: Encoder(serialize=lambda x: x.isoformat(), deserialize=lambda x: pandas.Timestamp(x)),
     pandas.DataFrame: Encoder(
         serialize=lambda x: jsonable_encoder(x.to_dict(orient='dict')),
-        deserialize=lambda x: x if isinstance(x,pandas.DataFrame) else _not_implemented(x, pandas.DataFrame),
+        deserialize=lambda x: x if isinstance(x, pandas.DataFrame) else _not_implemented(x, pandas.DataFrame),
     ),
 }
+
+
+def deserialize(value: Any, typ: Optional[Type]):
+    """
+    Deserialize a value into a given type.
+
+    Looks up the type in the encoder_registry and uses the deserializer to convert the value into the given type.
+
+    :param value: the value to deserialize
+    :param typ: the type to deserialize into
+    """
+    if isinstance(value, BaseTask):
+        return value
+
+    if typ is not None and typ in encoder_registry:
```

Comment:
more of a general question, if an if condition is used more than one time do we prefer having it repeated and larger statements or nested ifs?